.github/workflows/release.yml

```diff
@@ -0,0 +1,96 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  collect-modules:
+    runs-on: ubuntu-latest
+    outputs:
+      modules: ${{ steps.collect-modules.outputs.modules }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: collect modules
+        id: collect-modules
+        run: |
+          modules=$(find . -name .releaserc.yaml | \
+            xargs -I {} dirname {} | \
+            jq -R '[.]' | jq -s -c 'add')
+
+            echo "modules=$modules" >> $GITHUB_OUTPUT
+
+  release-modules:
+    needs: ["collect-modules"]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        module: ${{fromJson(needs.collect-modules.outputs.modules)}}
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: cloudsmith-io/cloudsmith-cli-action@v1.0.3
+        with:
+          api-key: ${{ secrets.CLOUDSMITH_API_KEY }}
+
+      - name: Fetch tags
+        run: git fetch --prune --tags
+
+      - name: collect module info
+        id: module-info
+        run: |
+          module_path="${{ matrix.module }}"
```

Comment:
Let's set the env variables `module_path` via a dedicated `env` key at the job level